US81119 - Change to behave like LMS search inputs

ryantmer opened the pull request:
- Remove button from left side of field
- Search icon is right-aligned in search field
- When user searches, the icon changes to X
- When X is clicked, search results and field are cleared

Also:

- Fix up demo to be a bit more coherent
- Remove overly-generic ID, as it may cause problems with Shady DOM

ctwomey1 commented on d2l-search-widget.html:
```diff
@@ -187,14 +174,44 @@
+			},
+			_setClearIcon: function() {
+				this.$$('d2l-icon').setAttribute('icon', 'd2l-tier1:close-default');
+				this.$$('button').setAttribute('aria-label', this.localize('search.clear'));
```
Same thing as https://github.com/Brightspace/d2l-my-courses-ui/pull/274 query selectors seem a bit too general.  Should be set to ids or just more specific to avoid breaks on change

ctwomey1 commented on d2l-search-widget.html:
```diff
@@ -232,30 +258,28 @@
-				if (response.detail.status === 200 || response.detail.status === 304) {
+			_onSearchResponse: function(e) {
+				if (e.detail.status === 200 || e.detail.status === 304) {
 					var parser = document.createElement('d2l-siren-parser');
```
I know this isn't your change, but do we need to recreate the parser every response?

ryantmer replied:
Nah, legit - been doing it elsewhere, might as well while I'm here.

ryantmer commented:
Removed the re-creation of `d2l-siren-parser` and slightly updated the CSS selectors - staying away from IDs since they can be bad with Shady DOM, but `this.$$` + the given selectors is specific enough in this case I think. If things changed to any degree, the tests would highlight the problem anyway 😜 (On that note, added checks for `aria-label` since forgot that initially.)
